crates/telio-pq/src/conn.rs

```diff
@@ -27,29 +27,30 @@ impl ConnKeyRotation {
         let request_retry = Duration::from_secs(features.handshake_retry_interval_s as _);
 
         let task = async move {
-            let mut interval = telio_utils::interval(request_retry);
+            let mut retry_interval = telio_utils::interval(request_retry);
 
             let proto::KeySet {
                 mut wg_keys,
                 pq_secret,
             } = loop {
-                interval.tick().await;
+                retry_interval.tick().await;
 
                 // Dylint is unhappy about the `fetch_keys` future size
                 // and asks for using `Box::pin` to move it on the heap
-                match Box::pin(super::proto::fetch_keys(
+                let fetch_keys = Box::pin(super::proto::fetch_keys(
                     &socket_pool,
                     addr,
                     &wg_secret,
                     &peer,
-                ))
-                .await
-                {
-                    Ok(keys) => {
+                ));
+
+                match tokio::time::timeout(request_retry, fetch_keys).await {
+                    Ok(Ok(keys)) => {
                         telio_log_debug!("PQ keys fetched");
                         break keys;
                     }
-                    Err(err) => telio_log_warn!("Failed to fetch PQ keys: {err}"),
+                    Ok(Err(err)) => telio_log_warn!("Failed to fetch PQ keys: {err}"),
+                    Err(_timeout) => telio_log_warn!("Failed to fetch PQ keys: TIMEOUT"),
```

Comment:
very minor - could you implement the timeout value in seconds in the error?

Reply:
Do you mean like printing the timeout?

Reply:
I've added the print